### DISCUSSION TITLE
RadForce test failure

### DISCUSSION THREAD

**Discussion by haakoan** [quokka-astro/quokka#15] [Q&A] [answered]
The test RadForce, run with make test, fails soon after starting. All other tests run without issue.

The test gives the following output and error:
```

MPI initialized with 1 MPI processes
MPI initialized with thread support level 0
AMReX (21.09-16-gd187e374f11c) initialized

Coarse STEP 1 at t = 0 (0%) starts ...
terminate called after throwing an instance of 'std::runtime_error'
  what():  Hydro states invalid (..../quokka/src/RadhydroSimulation.hpp:363)
Aborted (core dumped)
```

Is there a reason why this particular test would fail, while all others pass?

Thanks in advance for any help.

**Comment by BenWibking** [comment]
All tests pass when I run them. Can you copy the output from cmake so I can see exactly what build options and compiler options you are using?

**Reply by BenWibking** [reply]
I noticed the AMReX version is not the current version used in the repository. It might be useful to do `git pull` and `git submodule update`. The version string for the current version of AMReX included with Quokka should read `AMReX (21.10-24-gb60a773d6c87)`.

**Comment by haakoan** [comment, marked as answer]
After many tries to build the code on my local HPC system I got it working, but I lost the output of cmake for the version of my attempts that produced this issue. I think you are right and that it had to do with AMReX. I think the problem was due to the set up of the packages and environmental variables of the HPC system. 

I redid everything, with GCC, and got it running.

I had to manually set some of the CMAKE options. I think HPC users might run into a similar issues, the cmake does not always find the right compilers/python installations if the system has a complicated set up.

**Reply by BenWibking** [reply]
Were you using the Intel compiler by chance? Issue #5 notes that the `classic' Intel compiler (icc or icpc) generates incorrect machine code. If that was the issue, I will put a warning in the installation and readme about this.

**Reply by haakoan** [reply]
Yes I was. I saw this issue and it is why, together with a rather annoying setup of modules on my HPC system, that I changed to GCC.  I find it amazing that the intel compiler actually does this. On that note, have you compared the performance of GCC and Intel?

**Reply by BenWibking** [reply]
The 'new' intel compilers (icx, icpx) are about 5% faster than gcc in my tests.
